MinecraftClient/Protocol/Handlers/Protocol18.cs

```diff
@@ -535,6 +540,130 @@ internal bool HandlePacket(int packetID, List<byte> packetData)
                         SendPacket(PacketOutgoingType.ResourcePackStatus, dataTypes.ConcatBytes(responseHeader, dataTypes.GetVarInt(3))); //Accepted pack
                         SendPacket(PacketOutgoingType.ResourcePackStatus, dataTypes.ConcatBytes(responseHeader, dataTypes.GetVarInt(0))); //Successfully loaded
                         break;
+                    case PacketIncomingType.SpawnEntity:
+                        int EntityID = dataTypes.ReadNextVarInt(packetData);
```

Comment:
Handle entity-related packets only if terrain is enabled (or entity handling is enabled if such setting is added)